### ISSUE TITLE
Build broken

### ISSUE THREAD

**Issue by peria** [puyoai/puyoai#22]
I'm not sure what is the critical point, but it is sure that I cannot build programs after Munetoshi's commit [`851fea9083f4042d62a3208f10aeb89697a8791e`](https://github.com/puyoai/puyoai/commit/851fea9083f4042d62a3208f10aeb89697a8791e).
Could you check it, Munetoshi?

```
-- Configuring done
-- Generating done
-- Build files have been written to: /home/peria/github/puyo/ai/build
[6/155] Building CXX object core/CMakeFiles/puyoai_core.dir/field_pretty_printer.cc.o
FAILED: /usr/bin/clang++    -DDATA_DIR=\"/home/peria/github/puyo/ai/src/../data\" -DTESTDATA_DIR=\"/home/peria/github/puyo/ai/src/../testdata\" -DSRC_DIR=\"/home/peria/github/puyo/ai/src\" -Wno-error=sign-compare -Wno-error=missing-field-initializers -Wno-error=deprecated-declarations -Wall -Wextra -std=c++11 -std=c++11 -Werror -Wall -Wextra -Wno-error=unused-variable -Wno-error=unused-parameter -Wno-error=missing-field-initializers -fsanitize=address -O2 -g -I/home/peria/github/puyo/ai/src/. -I/home/peria/github/puyo/ai/src/third_party/gtest-1.7.0/include -I/home/peria/github/puyo/ai/src/third_party/gtest-1.7.0 -I/home/peria/github/puyo/ai/src/third_party/jsoncpp -I/home/peria/github/puyo/ai/src/third_party/tinytoml/include -I/usr/local/include -MMD -MT core/CMakeFiles/puyoai_core.dir/field_pretty_printer.cc.o -MF "core/CMakeFiles/puyoai_core.dir/field_pretty_printer.cc.o.d" -o core/CMakeFiles/puyoai_core.dir/field_pretty_printer.cc.o -c /home/peria/github/puyo/ai/src/core/field_pretty_printer.cc
In file included from /home/peria/github/puyo/ai/src/core/field_pretty_printer.cc:1:
In file included from /home/peria/github/puyo/ai/src/./core/field_pretty_printer.h:5:
In file included from /usr/lib/gcc/x86_64-linux-gnu/4.8/../../../../include/c++/4.8/string:41:
In file included from /usr/lib/gcc/x86_64-linux-gnu/4.8/../../../../include/c++/4.8/bits/allocator.h:46:
In file included from /usr/lib/gcc/x86_64-linux-gnu/4.8/../../../../include/x86_64-linux-gnu/c++/4.8/bits/c++allocator.h:33:
/usr/lib/gcc/x86_64-linux-gnu/4.8/../../../../include/c++/4.8/ext/new_allocator.h:93:7: error: multiple overloads of 'address' instantiate to the same signature 'const_pointer (const_reference) const noexcept'
      address(const_reference __x) const _GLIBCXX_NOEXCEPT
      ^
/usr/lib/gcc/x86_64-linux-gnu/4.8/../../../../include/c++/4.8/bits/allocator.h:92:29: note: in instantiation of template class '__gnu_cxx::new_allocator<const Position>' requested here
    class allocator: public __allocator_base<_Tp>
                            ^
/usr/lib/gcc/x86_64-linux-gnu/4.8/../../../../include/c++/4.8/bits/alloc_traits.h:90:24: note: in instantiation of template class 'std::allocator<const Position>' requested here
      typedef typename _Alloc::value_type value_type;
                       ^
/usr/lib/gcc/x86_64-linux-gnu/4.8/../../../../include/c++/4.8/ext/alloc_traits.h:123:5: note: in instantiation of template class 'std::allocator_traits<std::allocator<const Position> >' requested here
  : std::allocator_traits<_Alloc>
    ^
/usr/lib/gcc/x86_64-linux-gnu/4.8/../../../../include/c++/4.8/bits/stl_vector.h:74:35: note: in instantiation of template class '__gnu_cxx::__alloc_traits<std::allocator<const Position> >' requested here
      typedef typename __gnu_cxx::__alloc_traits<_Alloc>::template
                                  ^
/usr/lib/gcc/x86_64-linux-gnu/4.8/../../../../include/c++/4.8/bits/stl_vector.h:210:30: note: in instantiation of template class 'std::_Vector_base<const Position, std::allocator<const Position> >' requested here
    class vector : protected _Vector_base<_Tp, _Alloc>
                             ^
/usr/lib/gcc/x86_64-linux-gnu/4.8/../../../../include/c++/4.8/bits/stl_vector.h:161:9: note: in instantiation of template class 'std::vector<const Position, std::allocator<const Position> >' requested here
                      - this->_M_impl._M_start); }
                      ^
/usr/lib/gcc/x86_64-linux-gnu/4.8/../../../../include/c++/4.8/bits/stl_vector.h:248:7: note: in instantiation of member function 'std::_Vector_base<std::vector<const Position, std::allocator<const Position> >, std::allocator<std::vector<const Position, std::allocator<const Position> > > >::~_Vector_base' requested here
      vector()
      ^
/home/peria/github/puyo/ai/src/./core/rensa_result.h:78:2: note: in instantiation of member function 'std::vector<std::vector<const Position, std::allocator<const Position> >, std::allocator<std::vector<const Position, std::allocator<const Position> > > >::vector' requested here
        RensaVanishingPositionResult() {
        ^
/usr/lib/gcc/x86_64-linux-gnu/4.8/../../../../include/c++/4.8/ext/new_allocator.h:89:7: note: previous declaration is here
      address(reference __x) const _GLIBCXX_NOEXCEPT
      ^
/usr/lib/gcc/x86_64-linux-gnu/4.8/../../../../include/c++/4.8/ext/new_allocator.h:110:9: error: no matching function for call to 'operator delete'
      { ::operator delete(__p); }
        ^~~~~~~~~~
/usr/lib/gcc/x86_64-linux-gnu/4.8/../../../../include/c++/4.8/bits/stl_vector.h:174:12: note: in instantiation of member function '__gnu_cxx::new_allocator<const Position>::deallocate' requested here
          _M_impl.deallocate(__p, __n);
                  ^
/usr/lib/gcc/x86_64-linux-gnu/4.8/../../../../include/c++/4.8/bits/stl_vector.h:160:9: note: in instantiation of member function 'std::_Vector_base<const Position, std::allocator<const Position> >::_M_deallocate' requested here
      { _M_deallocate(this->_M_impl._M_start, this->_M_impl._M_end_of_storage
        ^
/usr/lib/gcc/x86_64-linux-gnu/4.8/../../../../include/c++/4.8/bits/stl_vector.h:414:7: note: in instantiation of member function 'std::_Vector_base<const Position, std::allocator<const Position> >::~_Vector_base' requested here
      ~vector() _GLIBCXX_NOEXCEPT
      ^
/usr/lib/gcc/x86_64-linux-gnu/4.8/../../../../include/c++/4.8/bits/stl_construct.h:93:19: note: in instantiation of member function 'std::vector<const Position, std::allocator<const Position> >::~vector' requested here
    { __pointer->~_Tp(); }
                  ^
/usr/lib/gcc/x86_64-linux-gnu/4.8/../../../../include/c++/4.8/bits/stl_construct.h:103:11: note: in instantiation of function template specialization 'std::_Destroy<std::vector<const Position, std::allocator<const Position> > >' requested here
            std::_Destroy(std::__addressof(*__first));
                 ^
/usr/lib/gcc/x86_64-linux-gnu/4.8/../../../../include/c++/4.8/bits/stl_construct.h:127:2: note: in instantiation of function template specialization 'std::_Destroy_aux<false>::__destroy<std::vector<const Position, std::allocator<const Position> > *>' requested here
        __destroy(__first, __last);
        ^
/usr/lib/gcc/x86_64-linux-gnu/4.8/../../../../include/c++/4.8/bits/stl_construct.h:151:7: note: in instantiation of function template specialization 'std::_Destroy<std::vector<const Position, std::allocator<const Position> > *>' requested here
      _Destroy(__first, __last);
      ^
/usr/lib/gcc/x86_64-linux-gnu/4.8/../../../../include/c++/4.8/bits/stl_vector.h:415:14: note: in instantiation of function template specialization 'std::_Destroy<std::vector<const Position, std::allocator<const Position> > *, std::vector<const Position, std::allocator<const Position> > >' requested here
      { std::_Destroy(this->_M_impl._M_start, this->_M_impl._M_finish,
             ^
/home/peria/github/puyo/ai/src/./core/rensa_result.h:78:2: note: in instantiation of member function 'std::vector<std::vector<const Position, std::allocator<const Position> >, std::allocator<std::vector<const Position, std::allocator<const Position> > > >::~vector' requested here
        RensaVanishingPositionResult() {
        ^
/usr/lib/gcc/x86_64-linux-gnu/4.8/../../../../include/c++/4.8/new:95:6: note: candidate function not viable: no known conversion from 'pointer' (aka 'const Position *') to 'void *' for 1st argument; take the address of the argument with &
void operator delete(void*) _GLIBCXX_USE_NOEXCEPT
     ^
/usr/lib/gcc/x86_64-linux-gnu/4.8/../../../../include/c++/4.8/new:103:6: note: candidate function not viable: requires 2 arguments, but 1 was provided
void operator delete(void*, const std::nothrow_t&) _GLIBCXX_USE_NOEXCEPT
     ^
/usr/lib/gcc/x86_64-linux-gnu/4.8/../../../../include/c++/4.8/new:115:13: note: candidate function not viable: requires 2 arguments, but 1 was provided
inline void operator delete  (void*, void*) _GLIBCXX_USE_NOEXCEPT { }
            ^
```

**Comment by peria** [comment]
Fixed with 051f97ae4997effae9e341ec32d3e9d63e96703a

**Comment by munetoshi** [comment]
For the record.
With @peria's environment, operator delete for elements is needed on vector deletion, so 
elements in RensaVanishingPositionResult::basePuyosErasedAt_ and fallingPuyosErasedAt_ should not be const.
